src/simulation.cpp

```diff
@@ -533,11 +533,15 @@ void initialize_history(Particle& p, int64_t index_source)
     auto site = sample_external_source(&seed);
     p.from_source(&site);
   }
+
   p.current_work() = index_source;
 
   // set identifier for particle
   p.id() = simulation::work_index[mpi::rank] + index_source;
 
+  // set particle history start weight
+  p.wgt0(p.wgt());
```

Comment:
We should set the initial particle weight in the `Particle::from_source` method instead of here in the same fashion that `wgt` and `wgt_last` are set.

Reply:
I tried what was mentioned, but it didn't work:
` p.wgt0(p.wgt());`

The legacy one does:
`p.wgt0() = p.wgt();`